Objective-C/CBLDatabase.mm

```diff
@@ -235,6 +235,12 @@ - (BOOL) purgeDocument: (CBLDocument*)document error: (NSError**)error {
                 if (!transaction.commit())
                     return convertError(transaction.error(), error);
                 
+                // Reload c4doc, still preserve the document data:
```

Comment:
How can the doc be reloaded after it's been purged? It should be completely gone from the database. If not, something's wrong...

Reply:
Currently the mostExist is set to false. It will get a new doc.
```
auto newDoc = c4doc_get(_c4db, bDocID, false, &err);
```

Reply:
Or probably I could just set the c4doc to nil? This is probably the same thing.